src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -713,25 +716,26 @@ def filter(self, col: Expression) -> "SelectStatement":
 
         return new
 
-    def sort(self, cols: List[Expression]) -> "SelectStatement":
+    def sort(self, cols: Sequence[Expression]) -> "SelectStatement":
```

Comment:
@sfc-gh-stan i see your pain here, mypy is converting List -> list, Dict -> dict, but you can not use list or dict for 3.8 at this moment. This seems start changing your signature, not sure if it is a good thing or not.
so one way to skip the check here is using typing.List, or typing.Dict, and we can start change it  to list or dict after 3.8 is deprecated.
i haven't figure out a way to ask mypy to skip the check yet